code/espurna/curtain_kingart.ino

```diff
@@ -0,0 +1,195 @@
+for the accurate control when manually controlling the device via the touch panel.
+
+After installing the switch and connecting the switch for the very first time:
+- First, it will automatically close the Cover/Shutter/Blind/Curtain to the maximum.
+- Press and hold the touch interface pause button for around 4 seconds until the red background
+  led lights up and starts blinking. Then, press the open touch button so start the opening process.
+- When cover is fully open, press the Open or Close button to stop the timer and save the calculated run time.
+
+To configure the device:
+- Press up/down for 5 seconds to bring device into AP mode. After pressing up/down again, device will restart in normal mode.
+*/
+
+#if KINGART_CURTAIN_SUPPORT
+
+#include <TimeLib.h> // now() definition
+
+#ifndef KINGART_CURTAIN_PORT
+#define KINGART_CURTAIN_PORT         Serial      // Hardware serial port by default
+#endif
+
+#ifndef KINGART_CURTAIN_BUFFER_SIZE
+#define KINGART_CURTAIN_BUFFER_SIZE  100         // Local UART buffer size
+#endif
+
+#define KINGART_CURTAIN_TERMINATION  '\e'        // Termination character after each message
+#define KINGART_CURTAIN_BAUDRATE     19200       // Serial speed is fixed for the device
+  
+char _KACurtainBuffer[KINGART_CURTAIN_BUFFER_SIZE];
+bool _KACurtainNewData = false;
+  
+// -----------------------------------------------------------------------------
+// Private
+// -----------------------------------------------------------------------------
+
+void _KACurtainSend(const char * tx_buffer) {
+    KINGART_CURTAIN_PORT.print(tx_buffer);
+    KINGART_CURTAIN_PORT.print(KINGART_CURTAIN_TERMINATION);
+    KINGART_CURTAIN_PORT.flush();
+}
+  
+void _KACurtainReceiveUART() {
+    static unsigned char ndx = 0;
+    while (KINGART_CURTAIN_PORT.available() > 0 && !_KACurtainNewData) {
+        char rc = KINGART_CURTAIN_PORT.read();
+        if (rc != KINGART_CURTAIN_TERMINATION) {
+            _KACurtainBuffer[ndx] = rc;
+            if (ndx < KINGART_CURTAIN_BUFFER_SIZE - 1) ndx++;
+        } else {
+            _KACurtainBuffer[ndx] = '\0';
+            _KACurtainNewData = true;
+            ndx = 0;
+        }
+    }
+}
+  
+void _KACurtainResult() {
+    if (_KACurtainNewData) {
+
+        // Need to send confiramtion to the N76E003AT20 that message is received
+        _KACurtainSend("AT+SEND=ok");
+
+        // We don't handle "setclose" any other way, simply redirect payload value
+        const String buffer(_KACurtainBuffer);
+        #if MQTT_SUPPORT
+            int setclose_idx = buffer.indexOf("setclose");
+            if (setclose_idx > 0) {
+                auto position = buffer.substring(setclose_idx + strlen("setclose") + 2, buffer.length());
+                int leftovers = position.indexOf(',');
+                if (leftovers > 0) {
+                    position = position.substring(0, leftovers);
+                }
+                mqttSend(MQTT_TOPIC_CURTAIN, String(position.toInt()));
+            }
```

Comment:
Quoting nodered cfg:
>     "func": "if (msg.payload.includes(\"sequence\")){\n    //Do nothing\n} else {\n    if (msg.payload.includes(\"setclose\")){\n        var output  = msg.payload.split(\":\");\n        var value   = output[2];\n        msg.payload = parseInt(value);\n        return msg;\n    }\n}\n",


Send number instead of sending raw payload to the `curtain` topic